scipy/stats/tests/test_stats.py

```diff
@@ -3734,25 +3719,17 @@ def test_skew_propagate_nan(self, xp):
     def test_skew_constant_value(self, xp):
         # Skewness of a constant input should be NaN (gh-16061)
         a = xp.asarray([-0.27829495]*10)  # xp.repeat not currently available
-        warn_ctx = (
-            contextlib.nullcontext() if is_lazy_array(a)
-            else pytest.warns(RuntimeWarning, match="Precision loss occurred..."))
 
-        with warn_ctx:
+        with eager_warns(a, RuntimeWarning, match="Precision loss occurred"):
             xp_assert_equal(stats.skew(a), xp.asarray(xp.nan))
-        with warn_ctx:
```

Comment:
This changes the test, but I suppose we're not really testing the warning here. 👍 (We're not testing much, from the looks of it, but that can be considered separately.)